src/App.js

```diff
@@ -22,109 +22,98 @@ mapboxgl.accessToken = config.accessToken;
 export default function App() {
     const mapContainer = useRef(null);
     const map = useRef(null);
-    const scroller = scrollama();
-    const data = createLine(rawData, config.driveSmoothness);
-    const feature = {
+    const line = createLine(rawData, config.driveSmoothness);
+    const empty = {
         type: 'FeatureCollection',
-        features: [
-            {
-                type: 'Feature',
-                geometry: {
-                    type: 'LineString',
-                    coordinates: [],
-                },
-            },
-        ],
+        features: [{ type: 'Feature', geometry: { type: 'LineString', coordinates: [] } }],
     };
-    const [isVisible, setVisibility] = useState(false); 
+    const [isVisible, setVisibility] = useState(false);
 
     const handleChange = () => {
         setVisibility(!isVisible);
     }
 
+    // Init scrollama, destroying the previous instance to prevent leaks
+    let scroller = null;
+    const resetScrollama = () => {
+        if (!map.current) return
+        if (scroller) scroller.destroy()
+        scroller = scrollama()
+            .setup({ step: '.step', offset: 0.5, progress: true })
+            .onStepEnter(response => actions.enter(response, config, map.current))
+            .onStepProgress(response => actions.progress(response, config, map.current, line))
+            .onStepExit(response => actions.exit(response));
+    };
+
+    // Since the map emits the `load` event only after fetching tiles,
+    // we need to init scrollama and emit progress events earlier
+    // to make sure the map doesn't load tiles we aren't interested in.
+    // Interval is a hack, there's probably a better way.
+    // We will remove it once the map finished loading.
+    let scrollReset = setInterval(resetScrollama, 200);
```

Comment:
Not 100% if I understand this one: I assumed the map would fetch the tiles according to what is first defined in the `Map` object when created. Since the `zoom`, `center` being etc passed to `Map` are the first story point, don't we need these tiles? Or what other tiles would be loaded that we are not interested in?

Reply:
Yes, but padding (that we used to set later on, if I remember correctly) affects this, and also if the page was for some reason not in the beginning (it was scrolled for a while and then refreshed or reopened) it would first load the initial region, then, once loaded move to the current one, triggering a second round of loading